opm/simulators/wells/BlackoilWellModel_impl.hpp

```diff
@@ -544,58 +543,60 @@ namespace Opm {
 
                 const Well* well_ecl = wells_ecl_[index_well];
 
-                // A new WCON keywords can re-open a well that was closed/shut due to Physical limit
-                if ( wellTestState_.hasWell(well_name, WellTestConfig::Reason::PHYSICAL ) ) {
-                    // TODO: more checking here, to make sure this standard more specific and complete
-                    // maybe there is some WCON keywords will not open the well
-                    if (well_state_.effectiveEventsOccurred(w)) {
-                        if (wellTestState_.lastTestTime(well_name) == ebosSimulator_.time()) {
-                            // The well was shut this timestep, we are most likely retrying
-                            // a timestep without the well in question, after it caused
-                            // repeated timestep cuts. It should therefore not be opened,
-                            // even if it was new or received new targets this report step.
-                            well_state_.setEffectiveEventsOccurred(w, false);
-                        } else {
-                            wellTestState_.openWell(well_name);
+                if (allow_closing_opening_wells) {
```

Comment:
if `allow_closing_opening_wells` is `false`, does this mean, even a well is shut due to economic or physical reason, we still need to calculate the well potentials?

maybe it is possible for the wells shut due to economic reasons, while it can be hard for the wells shut due to physical reason. 

I think it is reasonable to assume that we set the well potentials to be zero or not trying to calculate the well potentials for the wells are shut.